rewriter/rewriter.cc

```diff
@@ -187,9 +187,9 @@ ast::TreePtr Rewriter::run(core::MutableContext ctx, ast::TreePtr tree) {
     // This AST flattening pass requires that we mutate the AST in a way that our previous DSL passes were not designed
     // around, which is why it runs all at once and is not expressed as a `patch` method like the other DSL passes. This
     // is a rare case: in general, we should *not* add new DSL passes here.
-    auto defaulted = DefaultArgs::run(ctx, std::move(ast));
-    auto flattened = Flatten::run(ctx, std::move(defaulted));
-    auto cleaned = Cleanup::run(ctx, std::move(flattened));
+    auto flattened = Flatten::run(ctx, std::move(ast));
+    auto defaulted = DefaultArgs::run(ctx, std::move(flattened));
+    auto cleaned = Cleanup::run(ctx, std::move(defaulted));
```

Comment:
This re-orders default args to come after flatten. That means it can make assumptions like all method defs have been extracted to the top level, which means that things like `private def foo(x = nil); end` is no longer hidden from the defaults args pass.